app/agents/JobOfferManagerAgent.py

```diff
@@ -145,3 +165,31 @@ async def init_recr(self, new_applications: list[ApplicationDetails]):
             self.agent.initRmentsBehav = InitRecruitments()
             self.agent.add_behaviour(self.agent.initRmentsBehav)
             await self.agent.initRmentsBehav.join()
+
+class TriggerAnalysis(spade.behaviour.OneShotBehaviour):
+    """
+    Sends analyze request to ApplicationAnalyzer
+    Protocols/Activities in GAIA (role ApplicationHandler): TriggerAnalysis
+    """
+
+    agent: JobOfferManagerAgent
+
+    async def run(self):
+        self.agent.logger.info("TriggerAnalysis behaviour run.")
+
+        to_analyze = self.agent.applications_to_analyze
+
+        for app in to_analyze:
+            msg = await self.agent.prepare_message(
+                f"{self.agent.analyzerJID}_{random.randint(1, self.agent.analyzers_count)}@{self.agent.config.server.name}",
+                "request",
+                "analyze",
+                MessageType.ANALYSIS_REQUEST,
+                [self.agent.job_offer_id, app.candidate_id]
+            )
```

Comment:
a może zamiast w pętli to wysyłać jeden komunikat z tymi wszystkimi do analizy?

Reply:
Pętla jest okej, bo musi być load balancing. Do zamknięcia

Reply:
Zamykam